SteamKit2/SteamKit2/Steam/Handlers/SteamMatchmaking/Lobby.cs

```diff
@@ -0,0 +1,383 @@
+                int numMembers, List<Member> members, float? distance, long? weight )
+            {
+                if ( members != null && members.Count != numMembers - 1 )
+                {
+                    throw new ArgumentException( "when members is non-null, members.Count must be equal to numMembers - 1" );
+                }
+
+                SteamID = steamId;
+                LobbyType = lobbyType;
+                LobbyFlags = lobbyFlags;
+                OwnerSteamID = ownerSteamId;
+                metadata = metadata ?? new Dictionary<string, string>();
+                MaxMembers = maxMembers;
+                NumMembers = numMembers;
+                members = members ?? new List<Member>();
+                Distance = distance;
+                Weight = weight;
+            }
+
+            /// <summary>
+            /// Return a new <see cref="Lobby"/> instance with equivalent data as this lobby, but is safe to
+            /// access on another thread.
+            /// </summary>
+            /// <returns>A new <see cref="Lobby"/> instance that is equivalent to this one.</returns>
+            public Lobby Clone()
+            {
+                return new Lobby(
+                    SteamID,
+                    LobbyType,
+                    LobbyFlags,
+                    OwnerSteamID,
+                    new Dictionary<string, string>( metadata ),
+                    MaxMembers,
+                    NumMembers,
+                    new List<Member>( members ), // We're not doing a deep copy as Member is never mutated.
+                    Distance,
+                    Weight
+                );
+            }
+
+            internal static byte[] EncodeMetadata( Dictionary<string, string> metadata )
+            {
+                using ( var ms = new MemoryStream() )
+                {
+                    using ( var writer = new BinaryWriter( ms ) )
+                    {
+                        byte[] header = { 0, 0 };
+                        byte[] footer = { 8, 8 };
+
+                        writer.Write( header );
+
+                        foreach ( var pair in metadata )
+                        {
+                            if ( pair.Value == null ) continue;
+                            writer.Write( ( byte )1 );
+                            writer.Write( UTF8.GetBytes( pair.Key ) );
+                            writer.Write( ( byte )0 );
+                            writer.Write( UTF8.GetBytes( pair.Value ) );
+                            writer.Write( ( byte )0 );
+                        }
+
+                        ;
+
+                        writer.Write( footer );
+                    }
+
+                    return ms.ToArray();
+                }
+            }
+
+            internal static Dictionary<string, string> DecodeMetadata( byte[] data )
+            {
+                Dictionary<string, string> metadata = new Dictionary<string, string>();
```

Comment:
Is there a reason you're avoiding `var`?

Reply:
Just a 3am oversight when refactoring this code out of my own private codebase back into SteamKit. Already fixed locally.